compiler/src/dmd/frontend.h

```diff
@@ -2576,2298 +1701,2223 @@ class DelegateExp final : public UnaExp
-    bool lowerIsLessThanUpper() const;
-    bool lowerIsLessThanUpper(bool v);
-    bool arrayop() const;
-    bool arrayop(bool v);
-private:
-    uint8_t bitFields;
-public:
-    SliceExp* syntaxCopy() override;
-    bool isLvalue() override;
-    Optional<bool > toBool() override;
     void accept(Visitor* v) override;
 };
 
-class StringExp final : public Expression
+class PrettyFuncInitExp final : public DefaultInitExp
 {
 public:
-    char postfix;
-    OwnedBy ownedByCtfe;
-    union
-    {
-        char* string;
-        char16_t* wstring;
-        char32_t* dstring;
-    };
-    size_t len;
-    uint8_t sz;
-    bool committed;
-    bool hexString;
-    enum : char { NoPostfix = 0u };
-
-    static StringExp* create(const Loc& loc, const char* s);
-    static StringExp* create(const Loc& loc, const void* string, size_t len);
-    bool equals(const RootObject* const o) const override;
-    size_t numberOfCodeUnits(int32_t tynto = 0) const;
-    void writeTo(void* dest, bool zero, int32_t tyto = 0) const;
-    char32_t getCodeUnit(size_t i) const;
-    StringExp* toStringExp() override;
-    int32_t compare(const StringExp* const se2) const;
-    Optional<bool > toBool() override;
-    bool isLvalue() override;
     void accept(Visitor* v) override;
 };
 
-class StructLiteralExp final : public Expression
+class ObjcClassReferenceExp final : public Expression
 {
 public:
-    StructDeclaration* sd;
-    Array<Expression* >* elements;
-    Type* stype;
-    union
-    {
-        void* sym;
-        StructLiteralExp* inlinecopy;
-    };
-    StructLiteralExp* origin;
-    uint8_t stageflags;
-    bool useStaticInit;
-    bool isOriginal;
-    OwnedBy ownedByCtfe;
-    static StructLiteralExp* create(const Loc& loc, StructDeclaration* sd, void* elements, Type* stype = nullptr);
-    bool equals(const RootObject* const o) const override;
-    StructLiteralExp* syntaxCopy() override;
+    ClassDeclaration* classDeclaration;
     void accept(Visitor* v) override;
 };
 
-class ThisExp : public Expression
-{
-public:
-    VarDeclaration* var;
-    ThisExp(const Loc& loc, const EXP tok);
-    ThisExp* syntaxCopy() override;
-    Optional<bool > toBool() override;
-    bool isLvalue() final override;
-    void accept(Visitor* v) override;
-};
-
-class SuperExp final : public ThisExp
+class ClassReferenceExp final : public Expression
 {
 public:
+    StructLiteralExp* value;
+    ClassDeclaration* originalClass();
+    int32_t getFieldIndex(Type* fieldtype, uint32_t fieldoffset);
+    int32_t findFieldIndexByName(VarDeclaration* v);
     void accept(Visitor* v) override;
 };
 
-class SymbolExp : public Expression
+class ThrownExceptionExp final : public Expression
 {
 public:
-    Declaration* var;
-    Dsymbol* originalScope;
-    bool hasOverloads;
+    ClassReferenceExp* thrown;
+    const char* toChars() const override;
     void accept(Visitor* v) override;
 };
 
-class SymOffExp final : public SymbolExp
+class Expression : public ASTNode
```

Comment:
So now for example all definitions referenced by this class are emitted before this...